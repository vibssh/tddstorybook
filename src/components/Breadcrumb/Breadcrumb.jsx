import React from "react";
import PropTypes from "prop-types";
import "./Breadcrumb.scss";

const propTypes = {
  options: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string,
      url: PropTypes.string,
      isActive: PropTypes.bool
    })
  ),
  current: PropTypes.string
};

const defaultPropTypes = {
  options: [],
  current: 'active'
};

const Breadcrumb = props => {
  const { options, current } = props;
  return (
    <nav aria-label="breadcrumb">
      <ol className="breadcrumb">
        {options.map((option,index) => {
          const activeClass = option.isActive === true ? current : '';
          return (
            <li className={`breadcrumb-list ${activeClass}`} key={index}>
              {option.isActive !== true && <a href={option.url} className={`breadcrumb-list-link`}>
                {option.title}
              </a>
            }
            {option.isActive === true && option.title } 
            </li>
          );
        })}
      </ol>
    </nav>
  );
};

Breadcrumb.propTypes = propTypes;
Breadcrumb.defaultPropTypes = defaultPropTypes;
export default Breadcrumb;
