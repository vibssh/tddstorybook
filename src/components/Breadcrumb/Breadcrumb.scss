//Local variable
$divider: '/';
$divider-color: #ddd;
$breadcrumb-color: #007bff;
$breadcrumb-active-color: #6c757d;

.breadcrumb {
  margin: 0;
  padding: 0;
  list-style: none;

  &-list {
    display: inline-block;
    padding: 5px;
    padding-right: 0;
    color: $breadcrumb-active-color;

    &:after {
      content: $divider;
      display: inline-block;
      padding-left: 5px;
      color: $divider-color;
    }

    &:last-child {
      &:after {
        display: none;
      }
    }
  }

  &-list-link {
    width: 100%;
    color: $breadcrumb-color;
    text-decoration: none;

    &.active {
      color: $breadcrumb-active-color;
    }
  }
}