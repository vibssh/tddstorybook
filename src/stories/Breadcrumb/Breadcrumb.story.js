import React from 'react';
import { storiesOf } from '@storybook/react';
import BreadCrumb from '../../components/Breadcrumb';

const Items = {
  options: [
    {title: 'Home', url:'#home', isActive: true},
    {title: 'About', url:'#about', isActive: false}
  ],
  current: 'active'
}

storiesOf('Breadcrumb', module)
.addWithJSX('Breadcrumb', () => (
  <BreadCrumb {...Items } />
))
