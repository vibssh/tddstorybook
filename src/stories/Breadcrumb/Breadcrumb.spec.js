import React from 'react';
import {shallow} from 'enzyme';

import Breadcrumb from '../../components/Breadcrumb';

const Items = {
  options: [
    {title: 'Home', url:'#home', isActive: true},
    {title: 'About', url:'#about', isActive: false}
  ],
  current: 'active'
}

describe('BreadCrumb', () => {
  let wrapper;
  describe('It should render Breadcrumb', () => {
    it('should not render anchor link if breadcrumb is active', () => {
      wrapper = shallow(<Breadcrumb {...Items}/>);
      expect(wrapper.debug()).toMatchSnapshot();
    })
  })
})