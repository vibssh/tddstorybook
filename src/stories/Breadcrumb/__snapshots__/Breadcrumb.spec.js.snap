// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadCrumb It should render Breadcrumb should not render anchor link if breadcrumb is active 1`] = `
"<nav aria-label=\\"breadcrumb\\">
  <ol className=\\"breadcrumb\\">
    <li className=\\"breadcrumb-list active\\">
      Home
    </li>
    <li className=\\"breadcrumb-list \\">
      <a href=\\"#about\\" className=\\"breadcrumb-list-link\\">
        About
      </a>
    </li>
  </ol>
</nav>"
`;
